test/unit/beats_pipeline_generator_test.rb

```diff
@@ -29,11 +29,27 @@ def test_copy_and_rename_pipeline
 
   def test_non_renamed_beats
     mapping = {
-      'field1' => { source_field: 'field1', destination_field: 'field1', rename: 'copy' },
-      'field2' => { source_field: 'field2', destination_field: nil, rename: 'copy' },
+      'field1+field1' => { source_field: 'field1', destination_field: 'field1', rename: 'copy' },
+      'field2+' => { source_field: 'field2', destination_field: nil, rename: 'copy' },
     }
     pl = generate_beats_pipeline(mapping)
     assert_equal([], pl, "No rename processor should be added when there's no rename to perform")
   end
 
+  def test_duplicate_source_fields_same_destination
+    mapping = {
+      'field1+field3' => { source_field: 'field1', destination_field: 'field3', rename: 'copy' },
+      'field2+field3' => { source_field: 'field2', destination_field: 'field3', rename: 'copy' },
+      'field4+field5' => { source_field: 'field4', destination_field: 'field5', rename: 'copy' },
+      'field4+field6' => { source_field: 'field4', destination_field: 'field6', rename: 'copy' },
+    }
+
+    pl = generate_beats_pipeline(mapping)
+
+    assert_equal(
+      {"copy_fields"=>{"fields"=>[{"from"=>"field1", "to"=>"field3"}, {"from"=>"field2", "to"=>"field3"}, {"from"=>"field4", "to"=>"field5"}, {"from"=>"field4", "to"=>"field6"}], "ignore_missing"=>true, "fail_on_error"=>false}},
```

Comment:
Thanks for adding a test for the new use case :-)

Could you make this structure multi-line, to make it more readable? E.g. each small `{"from"=>"field1", "to"=>"field3"}` hash could be on its own line.